packages/core/src/node/console-logger-server.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { LogLevel } from '../common/logger';
+import { LoggerWatcher } from '../common/logger-watcher';
+import { LogLevelCliContribution } from './logger-cli-contribution';
+import { ILoggerServer, ILoggerClient } from '../common/logger-protocol';
+
+const originalConsoleLog = console.log;
```

Comment:
i wonder how we can be sure that they are not rebound already?

Reply:
Good point. Do you have an idea for that? I do not yet. It worth checking it but I am afraid it is the same when one defines `undefined`, what can we do with that?

Reply:
But that would only happen if the module gets loaded twice, which is an erroneous state anyways. Of course, some other library could do the same trick, but that shouldn't matter much as we end up in some chain of responsibility (unclear order, though). So, in the end, there is a certain contract to `console.log`. As long as nobody breaks it should be fine.

Reply:
I am OK leaving as is but the below snippet does the trick. Does it worth to delete the console:

```js
console.log = (msg, args) => {
    console.warn('Custom console log:', msg, args);
}

console.log('foo', 'bar');
delete console.log;
console.log('foo', 'bar');
```


```bash
$ node main.js 
Custom console log: foo bar
foo bar
```

From here: https://stackoverflow.com/questions/7089443/restoring-console-log

Reply:
I don't think so. It is not really our business to decide the implementation of the final delegated to console.log. As long as it satisfies the contract it is fine.

Reply:
ok, i was not worrying about externals, but more that it works only because `console-logger-server.ts` is loaded before `logger.ts`, changing the order of imports somewhere will break it

Reply:
Ah, I ok. That sounds of course a little bit brittle.

Reply:
So do we leave as is or I should add `delete console.x`?